ca/certificates.go

```diff
@@ -323,6 +324,45 @@ func (rca *RootCA) ParseValidateAndSignCSR(csrBytes []byte, cn, ou, org string)
 	return cert, nil
 }
 
+// DecryptRootKeyPEM decrypts a root key in PEM format using passphrases stored
+// in environment variables
+func DecryptRootKeyPEM(privKeyPEM []byte) ([]byte, error) {
+	// Decode the PEM private key
+	keyBlock, _ := pem.Decode(privKeyPEM)
+	if keyBlock == nil {
+		return nil, errors.New("malformed private key")
+	}
+	// If this key is not encrypted, then we're done
+	if !x509.IsEncryptedPEMBlock(keyBlock) {
+		return privKeyPEM, nil
+	}
+
+	for _, envVar := range []string{PassphraseENVVar, PassphraseENVVarPrev} {
+		strPassphrase := os.Getenv(envVar)
+		if strPassphrase == "" {
+			continue
+		}
+		passphrase := []byte(strPassphrase)
+		unencryptedKeyDER, err := x509.DecryptPEMBlock(keyBlock, []byte(passphrase))
+		if err != nil {
+			continue
+		}
+		headers := make(map[string]string)
```

Comment:
We should have a function that parses arbitrary PEMs w/ arbitrary PEM headers, instead of doing normalization inside of functions like this.